src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -256,60 +260,245 @@ public AnalyzerOptions AnalyzerOptions
+                    return _lazyRazorDesignTimeOptions ??= new RazorDesignTimeAnalyzerConfigOptions(_projectState.LanguageServices.WorkspaceServices);
+                }
+
+                var filePath = GetEffectiveFilePath(documentState);
+                if (filePath == null)
+                {
+                    return null;
+                }
+
+                var options = GetOptionsForSourcePath(cache, filePath);
+                var workspace = _projectState._solutionServices.Workspace;
+
+                var legacyIndentationService = workspace.Services.GetService<ILegacyIndentationManagerWorkspaceService>();
+                if (legacyIndentationService == null)
+                {
+                    return options;
+                }
+
+                return new AnalyzerConfigWithInferredIndentationOptions(options, workspace, legacyIndentationService, documentState.Id);
+            }
 
             public override AnalyzerConfigOptions GetOptions(AdditionalText textFile)
             {
                 // TODO: correctly find the file path, since it looks like we give this the document's .Name under the covers if we don't have one
-                return GetOptionsForSourcePath(textFile.Path);
+                return GetOptionsForSourcePath(GetCache(), textFile.Path);
             }
 
-            public AnalyzerConfigOptions GetOptionsForSourcePath(string path)
-                => new DictionaryAnalyzerConfigOptions(_projectState._lazyAnalyzerConfigOptions.GetValue(CancellationToken.None).GetOptionsForSourcePath(path).AnalyzerOptions);
+            private static StructuredAnalyzerConfigOptions GetOptionsForSourcePath(in AnalyzerConfigOptionsCache cache, string path)
+                => cache.GetOptionsForSourcePath(path).ConfigOptions;
+
+            private string? GetEffectiveFilePath(DocumentState documentState)
+            {
+                if (!string.IsNullOrEmpty(documentState.FilePath))
+                {
+                    return documentState.FilePath;
+                }
+
+                // We need to work out path to this document. Documents may not have a "real" file path if they're something created
+                // as a part of a code action, but haven't been written to disk yet.
+
+                var projectFilePath = _projectState.FilePath;
+
+                if (documentState.Name != null && projectFilePath != null)
+                {
+                    var projectPath = PathUtilities.GetDirectoryName(projectFilePath);
+
+                    if (!RoslynString.IsNullOrEmpty(projectPath) &&
+                        PathUtilities.GetDirectoryName(projectFilePath) is string directory)
+                    {
+                        return PathUtilities.CombinePathsUnchecked(directory, documentState.Name);
+                    }
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Provides editorconfig options for Razor design-time documents.
```

Comment:
when do we need options on the razor design time documents?  If they're asking us for formatting in razor, they should be p[assing the tabs/spaces/size options to us.  If there are cases where we're not passing the options / using the passed in topions, is there a bug tracking it?
cc @NTaylorMullen

Reply:
Good question. I'm not entirely sure. I think there might cases when a Roslyn feature might trigger change in a Razor document.

Reply:
@davidwengier

Reply:
Unfortunately I forget. @allisonchou do you recall?